client/db/src/storage_cache.rs

```diff
@@ -218,6 +220,27 @@ impl<B: BlockT> Cache<B> {
 			self.modifications.clear();
 		}
 	}
+
+	fn add_modifications(&mut self, block_changes: BlockChanges<B::Header>) {
+		let insert_at = self.modifications.iter()
+			.enumerate()
+			.find(|(_, m)| m.number < block_changes.number)
+			.map(|(i, _)| i);
+
+		trace!("Inserting modifications at {:?}", insert_at);
+		if let Some(insert_at) = insert_at {
+			self.modifications.insert(insert_at, block_changes);
+		} else {
+			self.modifications.push_back(block_changes);
+		}
+	}
+
+	/// Returns if this is the first modification at the given block height.
+	///
+	/// If there already exists a modification for a higher block height, `false` is returned.
+	fn is_first_modification_at_block_height(&self, number: NumberFor<B>) -> bool {
+		self.modifications.get(0).map(|c| c.number < number).unwrap_or(true)
```

Comment:
I'd rather call it `has_no_modification_at_block_height`. `is_` implies that condition is applied to `self`, rather than the argument.